### PR TITLE
Block use of git commit

### LINKED ISSUE #39: Disable committing in favor of using git gud commit using git hooks
When users are using git gud, we want more control over what they're committing so that we can track their progress, so we should disable the git commit command. We can do this using the pre-commit git hook and exiting with a non-zero value. It can be anything, but let's use 1.

We'll also want a log message that tells the user what happend, so we can make a simple file that prints out a log message, then calls `exit(1)`

See `gitgud/hooks/` for examples

This is also useful reference, but you shouldn't need it here: https://git-scm.com/docs/githooks

### PR DESCRIPTION
**Pull request by divCurlF**
- Stop use of "git commit" and "git commit -amend"
- Git hooks are now executable
- "git gud commits" ignore git hooks.

fixes #39

### CODE REVIEW

**Comment by divCurlF** [comment]
In reference to #39

**Comment by bthayer2365** [comment]
I'll get to this tomorrow

File: gitgud/__main__.py [hunk 1]
```diff
@@ -148,10 +148,21 @@ def handle_start(self, args):
         python_exec = sys.executable.replace('\\', '/')  # Git uses unix-like path separators
 
         for git_hook_name, module_hook_name in all_hooks:
-            with open(os.path.join(self.file_operator.hooks_path, git_hook_name), 'w+') as hook_file:
+            path = os.path.join(self.file_operator.hooks_path, git_hook_name)
+            with open(path, 'w+') as hook_file:
                 hook_file.write('#!/bin/sh' + os.linesep)
-                hook_file.write('cat - | ' + python_exec + ' -m gitgud.hooks.' + module_hook_name + ' $1' +os.linesep)
-                hook_file.write('exit 0' + os.linesep)
+                hook_file.write('' + python_exec + ' -m gitgud.hooks.' + module_hook_name + os.linesep)
```
Comment (bthayer2365) [inline gitgud/__main__.py]: Can you change that $1 to a $@? You removed it. Your current code breaks other git hooks. See #78

File: gitgud/__main__.py [hunk 2]
```diff
@@ -148,10 +148,21 @@ def handle_start(self, args):
...
+                    "if [[ $? -eq 1 ]]" + os.linesep + "" \
+                    "then" + os.linesep + "" \
+                    "\t exit 1" + os.linesep+ "" \
+                    "fi" + os.linesep)
+
+            # Make the files executable
+
+            mode = os.stat(path).st_mode
+            mode |= (mode & 0o444) >> 2
+            os.chmod(path, mode)
```
[the remark this answers is not in the document]
Comment (bthayer2365) [reply gitgud/__main__.py]: If I'm reading this right, it seems like you get the status and then if there's read permissions, then you also grant execute permissions, is that right? The "&" checks for read permissions, ">> 2" turns it into execute permissions and then "|=" makes sure you're adding the permission, not taking it away. Is that right?

**Comment by bthayer2365** [comment]
@divCurlF Are you still working on this?

**Comment by benthayer** [comment]
Hey @divCurlF 

In the next two weeks, I'm trying to get Git Gud to a version where I can start running user tests and this is part of the feature set I'd like to have at that point. If you're still working on this, I'm happy to help, and if not, I can take over and get this integrated into the master branch

**Comment by divCurlF** [comment]
> 
> 
> Hey @divCurlF
> 
> In the next two weeks, I'm trying to get Git Gud to a version where I can start running user tests and this is part of the feature set I'd like to have at that point. If you're still working on this, I'm happy to help, and if not, I can take over and get this integrated into the master branch

Sorry @benthayer I suck at managing my time, the changes look pretty easy so if you just want them done and merged go for it.

**Comment by benthayer** [comment]
If you can find time to do this over the next week, go for it. Otherwise, I'll get started with this on Monday 12/16.

**Pull request merged**